### PR TITLE
Anchors for example blocks in documentation

### PR DESCRIPTION
**Pull request by behackl**
## List of Changes
This PR adds anchors to the example blocks.

## Motivation
This was discussed on Discord and makes it easier to point to particular examples.

## Explanation for Changes
It was actually a bit painful to implement this. I first wanted to switch away from custom-building this via raw html blocks, but unfortunately this is not so easy as ReST does not offer an alternative to specifying section headings without using the under-/overline notation.

For now, this should work nonetheless.

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

File: docs/source/manim_directive.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def run(self):
             shutil.copyfile(filesrc, destfile)
         elif save_as_gif:
             filename = f"{output_file}.gif"
-            filesrc = os.path.join(video_dir, qualitydir, filename)
+            filesrc = config.get_dir("video_dir") / filename
```
Comment (leotrs) [inline docs/source/manim_directive.py:232]: Stray from another PR?

File: docs/source/manim_directive.py [hunk 2]
```diff
@@ -276,18 +277,20 @@ def setup(app):
...
-    <video class="manim-video" controls loop autoplay src="./{{ output_file }}.mp4"></video>
+    <video id="{{ clsname_lowercase }}" class="manim-video" controls loop autoplay src="./{{ output_file }}.mp4"></video>
 {% elif save_as_gif %}
 .. image:: /{{ filesrc_rel }}
     :align: center
+    :name: {{ clsname_lowercase }}
 {% elif save_last_frame %}
 .. image:: /{{ filesrc_rel }}
     :align: center
+    :name: {{ clsname_lowercase }}
```
Comment (leotrs) [inline docs/source/manim_directive.py:288]: Question: I'm not following - why are these two necessary?

Comment (leotrs) [reply docs/source/manim_directive.py:288]: Oh are these the equivalent of adding "id" on a div?

File: docs/source/manim_directive.py [hunk 3]
```diff
@@ -276,18 +277,20 @@ def setup(app):
...
 {% endif %}
 {% if not hide_source %}
 .. raw:: html
 
-    <div class="example-header">{{ clsname }}</div>
+    <h5 class="example-header">{{ clsname }}<a class="headerlink" href="#{{ clsname_lowercase }}">¶</a></h5>
```
Comment (leotrs) [inline docs/source/manim_directive.py:293]: I'd rather the anchor go to the title, but the video also works I guess.

File: docs/source/manim_directive.py (hunk 1 above)
Comment (behackl) [reply docs/source/manim_directive.py:232]: Yes, most likely a simple leftover.

File: docs/source/manim_directive.py (hunk 2 above)
Comment (behackl) [reply docs/source/manim_directive.py:288]: Precisely. The `:name:` option unfortunately converts whatever is passed to lowercase, that's why I needed to add `clsname_lowercase`.

File: docs/source/manim_directive.py (hunk 3 above)
Comment (behackl) [reply docs/source/manim_directive.py:293]: I found to the video/image a bit more intuitive, then you don't have to scroll up before seeing what the example is about. :-)

**Review by leotrs — APPROVED** [review]

**Review by naveen521kk — APPROVED** [review]
LGTM!

**Pull request merged**